addon/lib/ext/string.js

```diff
@@ -4,7 +4,7 @@ import {
   singularize
 } from '../system/string';
 
-if (Ember.EXTEND_PROTOTYPES === true || Ember.EXTEND_PROTOTYPES.String) {
+if (Ember.EXTEND_PROTOTYPES === true && Ember.EXTEND_PROTOTYPES.String) {
```

Comment:
i don't think this is correct, as  `Ember.EXTEND_PROTOTYPES  = { String: true }` will now fail as it will fail the precondition of `Ember.EXTEND_PROTOTYPES === true`